src/controllers/revision.ts

```diff
@@ -0,0 +1,478 @@
+            next(new UnknownException('errors.cube_create_error'));
+            return;
+        }
+    }
+    const downloadFile = await outputCube(cubeFile, lang, DuckdbOutputType.Csv);
+    fs.unlinkSync(cubeFile);
+    const downloadStream = fs.createReadStream(downloadFile);
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    res.writeHead(200, { 'Content-Type': '\ttext/csv' });
+    downloadStream.pipe(res);
+
+    // Handle errors in the file stream
+    downloadStream.on('error', (err) => {
+        logger.error('File stream error:', err);
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        res.writeHead(500, { 'Content-Type': 'text/plain' });
+        fs.unlinkSync(downloadFile);
+        res.end('Server Error');
+    });
+
+    // Optionally listen for the end of the stream
+    downloadStream.on('end', () => {
+        fs.unlinkSync(downloadFile);
+        logger.debug('File stream ended');
+    });
+};
+
+export const downloadRevisionCubeAsParquet = async (req: Request, res: Response, next: NextFunction) => {
+    const dataset = res.locals.dataset;
+    const revision = dataset.revisions.find((revision: Revision) => revision.id === req.params.revision_id);
+    const lang = req.language.split('-')[0];
+    if (!revision) {
+        next(new UnknownException('errors.no_revision'));
+        return;
+    }
+    let cubeFile: string;
+    if (revision.onlineCubeFilename) {
+        const dataLakeService = new DataLakeService();
+        const fileBuffer = await dataLakeService.getFileBuffer(revision.onlineCubeFilename, dataset.id);
+        cubeFile = tmp.tmpNameSync({ postfix: '.duckdb' });
+        fs.writeFileSync(cubeFile, fileBuffer);
+    } else {
+        try {
+            cubeFile = await createBaseCube(dataset, revision);
+        } catch (err) {
+            logger.error(`Something went wrong trying to create the cube with the error: ${err}`);
+            next(new UnknownException('errors.cube_create_error'));
+            return;
+        }
+    }
+    const downloadFile = await outputCube(cubeFile, lang, DuckdbOutputType.Parquet);
+    fs.unlinkSync(cubeFile);
+    const downloadStream = fs.createReadStream(downloadFile);
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    res.writeHead(200, { 'Content-Type': '\tapplication/vnd.apache.parquet' });
+    downloadStream.pipe(res);
+
+    // Handle errors in the file stream
+    downloadStream.on('error', (err) => {
+        logger.error('File stream error:', err);
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        res.writeHead(500, { 'Content-Type': 'text/plain' });
+        fs.unlinkSync(downloadFile);
+        res.end('Server Error');
+    });
+
+    // Optionally listen for the end of the stream
+    downloadStream.on('end', () => {
+        fs.unlinkSync(downloadFile);
+        logger.debug('File stream ended');
+    });
+};
```

Comment:
There are very similar handlers above for downloading cubes - do we need both sets?

Reply:
Yes... One relates to downloading the latest version of the cube... These relate to downloading a cube from a specific revision.  e.g. You're working on a revision which isn't published and want a preview of it or you need a historic revision of a cube.

You'll always need both... Sometimes you just want the latest version off the dataset.. publishers will mostly want one based on their current revision.  The current revision means it doesn't need to be published vs off the dataset where there *needs* to be a revision index to include it in the cube output.